overlord/configstate/configcore/homedirs.go

```diff
@@ -88,6 +82,10 @@ func configureHomedirsInAppArmorAndReload(homedirs []string, opts *fsOnlyContext
 
 	// Only reload profiles if it's during runtime
 	if opts == nil {
+		if _, err := apparmorSetupSnapConfineSnippets(); err != nil {
+			return err
+		}
```

Comment:
the comment before the if is not quite right now, because this bit is writing an apparmor snippet and not purely about reloading, I think it's ok not to write it early because the snippet is only useful once snaps are installed and not earlier than that but it needs to be clarified. We probably need a spread test that sets this via gadget defaults and sees that things work as expected for snaps installed during first boot

Reply:
otoh might that mostly not relevant because this option should work only classic, we don't have a concept of classic only options but maybe this should be one? cc @mvo5

Reply:
Are we refering just to `apparmorSetupSnapConfineSnippets` that we want to do only on classic?

Reply:
no, we mean to ignore (what we do for core only options on classic, for this we would need a new classicOnly flag) or error for the entire option on core.

Reply:
Is this somewhat what you had in mind? cd3f3b7